### PR TITLE
[Data Object] Support for composite indices

### LINKED ISSUE #5782: [Data Object] Support for composite indices

### PR DESCRIPTION
**Pull request by weisswurstkanone**
resolves #5782

### CODE REVIEW

**Comment by dpfaffenbauer** [comment]
does this also allow having a unique-index on multiple fields?

File: models/DataObject/Traits/CompositeIndexTrait.php [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            }
+        }
+
+        foreach ($drop as $key) {
+            $this->db->queryIgnoreError('ALTER TABLE `'.$table.'` DROP INDEX `'. $key.'`;');
+        }
+
+        foreach ($add as $key) {
+            $columnName = $newIndicesMap[$key];
+            $this->db->queryIgnoreError(
```
[the remark this answers is not in the document]
Comment (weisswurstkanone) [reply models/DataObject/Traits/CompositeIndexTrait.php]: done

**Comment by brusch** [comment]
@weisswurstkanone Stan is complaining :|

**Comment by weisswurstkanone** [comment]
Hi @dpfaffenbauer currently no but I think this would be a nice addition for the next release. If you also feel that way please create a seperate ticket. 👍 Thx!

**Pull request merged**